src/tree/ResolvedFunctionAppBase.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type ParsedSite } from "@microsoft/vscode-azext-azureappservice";
+import { nonNullProp, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import { type ResolvedAppResourceBase } from "@microsoft/vscode-azext-utils/hostapi";
+import { type FuncHostRequest } from "./IProjectTreeItem";
+import { type ContainerSite } from "./containerizedFunctionApp/ResolvedContainerizedFunctionAppResourceBase";
+
+export abstract class ResolvedFunctionAppBase implements ResolvedAppResourceBase {
```

Comment:
This is pretty bare bones still. I think there are quite a few more shared properties/functionality that you can add here.

Also abstract classes have the benefit of being able to define functions that are required on the class, but let each class that extends it implement it.

A good use case for this is something like `getApplicationSettings` where the parameters and the return result are the same, but the implementation is different between `ResolvedFunctionAppResource` and `ResolvedContainerizedFunctionAppResource`.

Reply:
Looks much better btw 🙂